src/linter-jscs.js

```diff
@@ -105,7 +108,11 @@ export default class LinterJSCS {
             this.fixString(editor);
           }
         }
-      });
+      }));
+      currentEditor.add(editor.onDidDestroy(() => {
+        currentEditor.dispose();
```

Comment:
With `const` we can be sure of that. Maybe it wouldn't work the same in cofeescript, I seem to recall something about differing scoping rules.